src/main/java/net/minecraftforge/fml/common/registry/ObjectHolderRef.java

```diff
@@ -99,23 +102,18 @@
 
     public boolean isValid()
     {
-        return isBlock || isItem;
+        return isValid;
     }
     public void apply()
     {
         Object thing;
-        if (isBlock)
+        if (isValid)
         {
-            thing = ForgeRegistries.BLOCKS.getValue(injectedObject);
-            if (thing == Blocks.AIR)
-            {
+            thing = registry.getValue(injectedObject);
+            if (registry.getRegistrySuperType() == Block.class && thing == Blocks.AIR) {
```

Comment:
This really has nothing to do with the block type.
Its just making sure we return a null instead of the default value from the registry.
Thus allowing people to check if things exist in the registry with a null check instead of air check.